docs/rom-embedding.md

```diff
@@ -0,0 +1,122 @@
+# ROM File Embedding Implementation Summary
+
+## Overview
+This implementation adds the ability to embed binary files into ROM during build time, with two optional files:
+1. `startup.dmp` - Startup package automatically loaded at boot
+2. `user_data` - User data file with address/size exposed via environment variables
+
+## Implementation Details
+
+### 1. Helper Script (`scripts/embed_binary.cmake`)
+- **Purpose**: Provides `embed_binary_file()` function to convert binary files into ELF object files
+- **How it works**: 
+  - Uses `objcopy` to convert binary data into ARM ELF object files
+  - Places data in custom sections (e.g., `.embedded.startup_dmp`)
+  - Relies on linker script to create proper symbols
+
+### 2. Build Configuration (`CMakeLists.txt`)
+- **New CMake variables**:
+  - `STARTUP_DMP_FILE` (optional) - Path to startup.dmp file
+  - `USER_DATA_FILE` (optional) - Path to user_data file
+- **Build process**:
+  - Checks if files exist
+  - Calls `embed_binary_file()` to create object files
+  - Links object files into final executable
+
+### 3. Linker Script (`linker/common.ld`)
+- **New sections added**:
+  - `.embedded.startup_dmp` - Contains startup.dmp data
+  - `.embedded.user_data` - Contains user_data data
+- **Symbols created** (for each section):
+  - `__startup_dmp_start`, `__startup_dmp_end`, `__startup_dmp_size`
+  - `__user_data_start`, `__user_data_end`, `__user_data_size`
+
+### 4. Runtime Loading (`src/main.c`)
+- **startup.dmp handling**:
+  - Checks if `__startup_dmp_size > 0`
+  - If yes, calls `Dmod_AddPackageBuffer()` to load the package
```

Comment:
to jest nieaktualne - no i w ogole glupoty tu piszesz, bo to juz automatycznie sie ładuje przecież podczas startu dmod-boot. Opisz to lepiej z perspektywy uzytkownika koncowego